src/main/java/com/aws/greengrass/ipc/SubscribeToCertificateUpdatesOperationHandler.java

```diff
@@ -0,0 +1,167 @@
+import static com.aws.greengrass.ipc.common.ExceptionUtil.translateExceptions;
+
+public class SubscribeToCertificateUpdatesOperationHandler
+        extends GeneratedAbstractSubscribeToCertificateUpdatesOperationHandler {
+    private static final Logger logger = LogManager.getLogger(SubscribeToCertificateUpdatesOperationHandler.class);
+    private static final String COMPONENT_NAME = "componentName";
+    private static final String NO_CERT_OPTIONS_ERROR = "Certificate options are required.";
+    private static final String NO_CERT_TYPE_ERROR = "Certificate type is required";
+    private static final String UNAUTHORIZED_ERROR = "Not Authorized";
+    private static final int RSA_KEY_LENGTH_FOR_SERVER_CERT = 4096;
+    private final String serviceName;
+    private final ClientDevicesAuthService clientDevicesAuthService;
+    private final CertificateManager certificateManager;
+    private KeyPair keyPair;
+    private final AtomicReference<Boolean> initialResponseSent = new AtomicReference<>(false);
+    private final AtomicReference<CertificateUpdateEvent> firstStreamingevent = new AtomicReference<>(null);
+
+    /**
+     * Constructor.
+     *
+     * @param context                  operation continuation handler
+     * @param clientDevicesAuthService client device auth service
+     */
+    public SubscribeToCertificateUpdatesOperationHandler(OperationContinuationHandlerContext context,
+                                                         ClientDevicesAuthService clientDevicesAuthService) {
+        super(context);
+        serviceName = context.getAuthenticationData().getIdentityLabel();
+        this.clientDevicesAuthService = clientDevicesAuthService;
+        this.certificateManager = clientDevicesAuthService.getCertificateManager();
+
+    }
+
+    @Override
+    protected void onStreamClosed() {
+        //TODO: Unsubscribe to cert updates.
+    }
+
+    @Override
+    public void afterHandleRequest() {
+        initialResponseSent.compareAndSet(false, true);
+        while (this.firstStreamingevent.get() != null && initialResponseSent.get()) {
+            this.sendStreamEvent(this.firstStreamingevent.get());
+            this.firstStreamingevent.set(null);
+        }
+    }
+
+    @SuppressWarnings("PMD.PreserveStackTrace")
+    @Override
+    public SubscribeToCertificateUpdatesResponse handleRequest(
+            SubscribeToCertificateUpdatesRequest subscribeToCertificateUpdatesRequest) {
+        return translateExceptions(() -> {
+            try {
+                clientDevicesAuthService.doAuthorizationForSubscribingToCertUpdates(serviceName);
+            } catch (AuthorizationException e) {
+                logger.atInfo().kv("error", e.getMessage()).log(UNAUTHORIZED_ERROR);
+                throw new UnauthorizedError(e.getMessage());
+            }
+            try {
+                String certificateType = getCertificateTypeFromCertificateOptions(
```

Comment:
don't convert to a string and then compare it to the enum. Just have `getCertificateTypeFromCertificateOptions` return the enum type directly.